nbconvert/exporters/qtpdf.py

```diff
@@ -0,0 +1,130 @@
+    template machinery, and then uses pyqtwebengine to create a pdf.
+    """
+    export_from_notebook = "PDF via HTML"
+
+    paginate = Bool(
+        True,
+        help="""
+        Split generated notebook into multiple pages.
+
+        If False, a PDF with one long page will be generated.
+
+        Set to True to match behavior of LaTeX based PDF generator
+        """
+    ).tag(config=True)
+
+    output_mimetype = "application/pdf"
+
+    @default('file_extension')
+    def _file_extension_default(self):
+        return '.pdf'
+
+    @default('template_name')
+    def _template_name_default(self):
+        return 'qtpdf'
+
+    @default('template_data_paths')
+    def _template_data_paths_default(self):
+        return jupyter_path("nbconvert", "templates", "qtpdf")
+
+    def _check_launch_reqs(self):
+        try:
+            from PyQt5.QtWidgets import QApplication
+            from PyQt5 import QtCore
+            from PyQt5.QtWebEngineWidgets import QWebEngineView, QWebEngineSettings
+            from PyQt5.QtGui import QPageLayout, QPageSize
+        except ModuleNotFoundError as e:
+            raise RuntimeError("PyQtWebEngine is not installed to support Qt PDF conversion. "
+                               "Please install `nbconvert[qtpdf]` to enable.") from e
+        return QApplication, QtCore, QWebEngineView, QWebEngineSettings, QPageLayout, QPageSize
+
+    def run_pyqtwebengine(self, html):
+        """Run pyqtwebengine."""
+
+        ext = ".html"
+        temp_file = tempfile.NamedTemporaryFile(suffix=ext, delete=False)
+        pdf_filename = f"{temp_file.name[:-len(ext)]}.pdf"
+        with temp_file:
+            temp_file.write(html.encode('utf-8'))
+
+        try:
+            QApplication, QtCore, QWebEngineView, QWebEngineSettings, QPageLayout, QPageSize = self._check_launch_reqs()
+
+            class Screenshot(QWebEngineView):
+                def __init__(self, app, paginate):
+                    super().__init__()
+                    self.app = app
+                    self.paginate = paginate
+                def capture(self, url, output_file):
+                    self.output_file = output_file
+                    self.load(QtCore.QUrl(url))
+                    self.loadFinished.connect(self.on_loaded)
+                    # Create hidden view without scrollbars
+                    self.setAttribute(QtCore.Qt.WA_DontShowOnScreen)
+                    self.page().settings().setAttribute(
+                        QWebEngineSettings.ShowScrollBars, False)
+                    self.page().pdfPrintingFinished.connect(
+                        lambda *args: self.app.exit())
+                    self.show()
+                    self.app.exec()
+                def on_loaded(self):
+                    self.size = self.page().contentsSize().toSize()
+                    self.resize(self.size)
+                    # Wait for resize
+                    QtCore.QTimer.singleShot(1000, self.take_screenshot)
```

Comment:
Here we wait for one second for the resize to take effect. I know it's not great but I couldn't find another way.